### PR TITLE
tsp, customize Base64UrlArrayBytesProperty.java to pass encode/bytes test

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Last remaining issue in https://github.com/Azure/autorest.java/pull/2202#issuecomment-1598294295

We skipped handling that corner case in codegen.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,15 +36,7 @@
...
     "/preprocessor/*.md",
-    "*/target/*-jar-with-dependencies.jar",
-    "/postprocessor/jdt-language-server/config_linux/*.ini",
-    "/postprocessor/jdt-language-server/config_mac/*.ini",
-    "/postprocessor/jdt-language-server/config_ss_linux/*.ini",
-    "/postprocessor/jdt-language-server/config_ss_mac/*.ini",
-    "/postprocessor/jdt-language-server/config_ss_win/*.ini",
-    "/postprocessor/jdt-language-server/config_win/*.ini",
-    "/postprocessor/jdt-language-server/features",
-    "/postprocessor/jdt-language-server/plugins"
```
Comment (weidongxu-microsoft) [inline package.json:47]: clean up

File: typespec-tests/Generate.ps1 [hunk 2]
```diff
@@ -27,14 +35,16 @@ function Generate($tspFile) {
...
-    Copy-Item -Path ./customization -Destination ./tsp-output/customization -Recurse -Force
+  if ($tspFile -match "tsp[\\/]naming.tsp$") {
+#     # since tsp-output directory will be cleaned up after each test tsp, we copy the customization
+#     # code into output directory from customization directory in tsp-output/customization directory just before
+#     # generating the code
+#     Copy-Item -Path ./customization -Destination ./tsp-output/customization -Recurse -Force
 
     # Add the customization-class option for Java emitter
-    $tspOptions = "--options=""@azure-tools/typespec-java.customization-class=customization/src/main/java/CustomizationTest.java"""
+    $tspOptions = "--options=""@azure-tools/typespec-java.customization-class=../customization/src/main/java/CustomizationTest.java"""
```
Comment (weidongxu-microsoft) [inline typespec-tests/Generate.ps1:45]: I've commented out the copy.

Using relative path (`../###`) works fine for me.

File: typespec-tests/customization/src/main/java/CustomizationEncodeBytes.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        PackageCustomization packageCustomization = customization.getPackage("com.encode.bytes.models");
+        ClassCustomization classCustomization = packageCustomization.getClass("Base64UrlArrayBytesProperty");
+
+        ConstructorCustomization constructorCustomization = classCustomization.getConstructor("Base64UrlArrayBytesProperty");
+        constructorCustomization.removeAnnotation("JsonCreator");
+        constructorCustomization = classCustomization.getConstructor("Base64UrlArrayBytesProperty");
+        constructorCustomization.addAnnotation("JsonCreator(mode=JsonCreator.Mode.DISABLED)");
+
+        classCustomization.addConstructor("@Generated private Base64UrlArrayBytesProperty() {}");
+    }
```
Comment (weidongxu-microsoft) [inline typespec-tests/customization/src/main/java/CustomizationEncodeBytes.java:31]: customization code

File: typespec-tests/src/main/java/com/encode/bytes/models/Base64UrlArrayBytesProperty.java [hunk 4]
```diff
@@ -27,11 +27,14 @@ public final class Base64UrlArrayBytesProperty {
...
      */
     @Generated
-    @JsonCreator
+    @JsonCreator(mode = JsonCreator.Mode.DISABLED)
     public Base64UrlArrayBytesProperty(@JsonProperty(value = "value") List<byte[]> value) {
         this.value = value.stream().map(el -> Base64Url.encode(el)).collect(java.util.stream.Collectors.toList());
     }
 
+    @Generated
+    private Base64UrlArrayBytesProperty() {}
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/encode/bytes/models/Base64UrlArrayBytesProperty.java:36]: result of customization
